### PR TITLE
Fix resource editing dialog crashing

### PR DESCRIPTION
**Pull request by likawind**
## Describe your changes and why you are making these changes
This PR fixes resource dialog crashing issue in edit mode. We also added edit mode support for all refactored dialogs by properly fetching the values on current resources, and disable requirements on certain fields (e.g. credentials).

## Related issue number (if any)
ENG-3050

## Loom demo (if any)
Tested on editing S3, WIP testing on more integrations

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/ui/common/src/components/resources/dialogs/azureDialog.tsx [hunk 1]
```diff
@@ -1,6 +1,9 @@
 import React from 'react';
 import * as Yup from 'yup';
 
+export const AzureDialog: React.FC = () => null;
+
+/*
```
Comment (agiron123) [inline src/ui/common/src/components/resources/dialogs/azureDialog.tsx:6]: This is placeholder for now, right?

**Review by agiron123 — APPROVED** [review]
Haven't had time to give this any manual testing yet, but code looks good otherwise.

File: src/ui/common/src/components/resources/dialogs/azureDialog.tsx (hunk 1 above)
Comment (likawind) [reply src/ui/common/src/components/resources/dialogs/azureDialog.tsx:6]: yup!

File: src/ui/common/src/components/resources/dialogs/dialog.tsx [hunk 2]
```diff
@@ -108,23 +115,24 @@ const ResourceDialog: React.FC<Props> = ({
     // We do this so we can collect name form inputs inside the same form context.
     const name = data.name;
     // remove the name key from data so pydantic doesn't throw error.
-    delete data.name;
+    const config = { ...data };
```
Comment (likawind) [inline src/ui/common/src/components/resources/dialogs/dialog.tsx:118]: @eunice-chan I've manually included  your change in https://github.com/aqueducthq/aqueduct/pull/1381 here, so feel free to close the PR!

**Pull request merged**